vpr/src/place/initial_placement.cpp

```diff
@@ -51,38 +51,247 @@ static vtr::vector<ClusterBlockId, t_block_score> assign_block_scores();
 //Sort the blocks according to how difficult they are to place, prior to initial placement
 static std::vector<ClusterBlockId> sort_blocks(const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
 
-//Sort the macros according to how difficult they are to place, prior to initial placement
-static std::vector<t_pl_macro> sort_macros(const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
+/*
+ * Look for one random legal position for a block that respects its floorplanning constraints
+ * Returns true if a valid location is found, false otherwise
+ * Returns location (loc) by reference
+ */
+static bool get_legal_placement_loc(PartitionRegion& pr, t_pl_loc& loc, t_logical_block_type_ptr block_type);
+
+/*
+ * A routine to look for a valid placement location for a (non-macro) block exhaustively once the maximum number of random locations
+ * have been tried. Returns true if the block is placed, false otherwise.
+ */
+static bool try_all_part_region_locations(ClusterBlockId blk_id, PartitionRegion& pr, t_logical_block_type_ptr block_type, enum e_pad_loc_type pad_loc_type);
```

Comment:
Comment inputs, esp. pad_loc_type as it is a bit mysterious.
Think about whether you should pass in block type as it can be found from blk_id.
Evaluate changing this to return a to_loc to see if the calling code would then be simpler / more logical.  
- ask for random loc
- ask for exhaustive loc
- place (and lock down if pad_loc_type == RANDOM)